### DISCUSSION TITLE
Automatically Reconnecting to Server

### DISCUSSION THREAD

**Discussion by ConnorNelson** [coder/code-server#4836] [Q&A]
Hello!

I am wondering what the best way to support automatically reconnecting to the server is. Basically, I want to be able to start the server, connect in the browser, interact with VS code in the browser, restart the server, and continue interacting without needing to reload the page.

When testing this use-case, I see the message,
```
[00:33:12] [172.17.0.1][5ae13c2d][ManagementConnection] Unknown reconnection token (never seen).
```
and the browser remains disconnected from VS code until I refresh the browser tab.

To further expand upon my use case, I have users that will be performing work in one docker container, and eventually they need to be migrated to a different container. I see a potential path forward that would involve keeping the server running and migrating that running process to the new namespace, but I am hoping maybe there is a simpler path forward.

Thanks!

**Comment by jsjoeio** [comment]
Hey Connor! 

Great question. My hunch says that if you reload the server running code-server that you'll have to reload the page 🤔 

Maybe there's a solution though. Let me ping a few teammates and see if I can get some help from folks with more Docker experience than me. cc @code-asher @bpmct @jawnsy 

P.S. I see you're from Arizona! Super cool seeing other Arizonians in the code-server community. Bear down 🐻⬇️

**Reply by code-asher** [reply]
Yeah unfortunately this is due to VS Code's design.  It would need to be refactored upstream.
